Marlin/src/HAL/HAL_STM32F1/persistent_store_flash.cpp

```diff
@@ -67,42 +66,29 @@ bool PersistentStore::write_data(int &pos, const uint8_t *value, size_t size, ui
     firstWrite = false;
   }
 
-  // First write full words
-  int i = 0;
-  int wordsToWrite = size / sizeof(uint16_t);
-  uint16_t* wordBuffer = (uint16_t *)value;
-  while (wordsToWrite) {
-    status = FLASH_ProgramHalfWord(pageBase + pos + (i * 2), wordBuffer[i]);
-    if (status != FLASH_COMPLETE) return true;
-    wordsToWrite--;
-    i++;
-  }
-
-  // Now, write any remaining single byte
-  const uint16_t odd = size & 1;
-  if (odd) {
-    uint16_t temp = value[size - 1];
-    status = FLASH_ProgramHalfWord(pageBase + pos + i, temp);
-    if (status != FLASH_COMPLETE) return true;
+  for (int i=0; i<size; i++) {
```

Comment:
Marlin/src/HAL/HAL_STM32F1/persistent_store_flash.cpp: In static member function 'static bool PersistentStore::write_data(int&, const uint8_t*, size_t, uint16_t*)':
Marlin/src/HAL/HAL_STM32F1/persistent_store_flash.cpp:69:18: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (int i=0; i<size; i++) {
                 ~^~~~~